packages/orama/src/components/algorithms.ts

```diff
@@ -1,12 +1,13 @@
 import { createError } from '../errors.js'
 import { TokenScore, BM25Params } from '../types.js'
+import { InternalDocumentID } from './internal-document-id-store.js';
 
 export function prioritizeTokenScores(arrays: TokenScore[][], boost: number, threshold = 1): TokenScore[] {
   if (boost === 0) {
     throw createError('INVALID_BOOST_VALUE')
   }
 
-  const tokenMap: Record<string, number> = {}
+  const tokenMap = new Map<InternalDocumentID, number>()
```

Comment:
are we able to build this map with some content?
For instance, `pop()` from the right entries list and use that to build the map initially.
This could reduce the `get` failure

Reply:
IDs could be duplicated, so I don't think populating this map first will be faster than just leaving it empty.

![image](https://github.com/oramasearch/orama/assets/12551007/528737c0-eb4f-4631-bbcb-bb68c105e3f4)

This is the behavior when keys are duplicated.